packages/autocomplete-plugin-query-suggestions/src/createQuerySuggestionsPlugin.ts

```diff
@@ -117,23 +117,28 @@ export function createQuerySuggestionsPlugin<
                     return querySuggestionsHits as any;
                   }
 
+                  let itemsWithCategoriesAdded = 0;
                   return querySuggestionsHits.reduce<
                     Array<AutocompleteQuerySuggestionsHit<typeof indexName>>
-                  >((acc, current, i) => {
+                  >((acc, current) => {
                     const items: Array<
                       AutocompleteQuerySuggestionsHit<typeof indexName>
                     > = [current];
 
-                    if (i <= itemsWithCategories - 1) {
-                      const categories = getAttributeValueByPath(
-                        current,
-                        Array.isArray(categoryAttribute)
-                          ? categoryAttribute
-                          : [categoryAttribute]
+                    if (itemsWithCategoriesAdded < itemsWithCategories) {
+                      const categories = (
+                        getAttributeValueByPath(
+                          current,
+                          Array.isArray(categoryAttribute)
+                            ? categoryAttribute
+                            : [categoryAttribute]
+                        ) || []
                       )
                         .map((x) => x.value)
                         .slice(0, categoriesPerItem);
 
+                      itemsWithCategoriesAdded += categories.length;
```

Comment:
Shouldn't it be incremented by 1 only?
If item A has 5 categories you'll only show categories for this item and not up to the limit defined by `itemsWithCategories`.

Reply:
just below you're pushing to item every category though?

Reply:
That part is working, but if you increment by the number of categories retrieved, you might prevent further query suggestions from having categories attached to them. If a customer wants 3 `itemsWithCategories`, you have to go into your conditional branch 3 times, which might not happen if you increment by more than 1.

Here is the current playground with `{ itemsWithCategories: 3, categoriesPerItem: 2 }`:

<img width="695" alt="CleanShot 2023-02-10 at 16 49 46@2x" src="https://user-images.githubusercontent.com/154633/218135193-beb10a27-9bbe-4dfd-a9e5-b2c8c86aaf9f.png">

And here is the working branch with the same settings:

<img width="681" alt="CleanShot 2023-02-10 at 16 49 29@2x" src="https://user-images.githubusercontent.com/154633/218135142-cf6f0879-0f94-4a78-aa22-0426f3c51c48.png">

Reply:
oh I see

Reply:
```suggestion
                      itemsWithCategoriesAdded++;
```